applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -224,7 +231,11 @@ def ComputeDeltaTime(self):
         pass
 
     def SaveRestart(self):
-        pass #one should write the restart file here
+        # Check could be integrated in the utility
+        # It is here intentionally, this way the utility is only created if it is actually needed!
+        if (self.settings["restart_settings"]["save_restart"].GetBool() == True):
```

Comment:
`if ( self.is_restarted() ): ` ??

Reply:
this is the opposite, self.is_restarted is true if you start from a restart file
this check is abt saving a restart file